plugin.video.mytv_bg/addon.py

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#imports
+from xbmcswift import Plugin, xbmc, xbmcaddon, xbmcplugin, xbmcgui, clean_dict
+from urllib2 import urlopen, HTTPError
+import xbmcgui
+from pyxbmct.addonwindow import *
+import sys
+import json
+import os.path
+from urllib2 import Request
+import urllib
+import urllib2
+
+__addon_name__		= xbmcaddon.Addon().getAddonInfo('name')
+__id__				= xbmcaddon.Addon().getAddonInfo('id')
+__lang__			= xbmcaddon.Addon().getLocalizedString
+__addon__ 			= xbmcaddon.Addon()
+__version__ 		= __addon__.getAddonInfo('version')
+__profile_path__ 	= xbmc.translatePath( __addon__.getAddonInfo('profile') ).decode("utf-8")
+__token_filepath__	= __profile_path__ + '/token.txt'
+
+
+THUMBNAIL_VIEW_IDS = {'skin.confluence': 500,
+                      'skin.aeon.nox': 551,
+                      'skin.confluence-vertical': 500,
+                      'skin.jx720': 52,
+                      'skin.pm3-hd': 53,
+                      'skin.rapier': 50,
+                      'skin.simplicity': 500,
+                      'skin.slik': 53,
+                      'skin.touched': 500,
+                      'skin.transparency': 53,
+                      'skin.xeebo': 55}
+
+# BEGIN # Plugin
+class Plugin_mod(Plugin):
+
+    def add_items(self, iterable, is_update=False, sort_method_ids=[],
+                  override_view_mode=False):
+        items = []
+        urls = []
+        for i, li_info in enumerate(iterable):
+            
+            items.append( 
+                self._make_listitem(**li_info)
+            )
+
+            if self._mode in ['crawl', 'interactive']:
+                print '[%d] %s%s%s (%s)' % (i + 1, '', li_info.get('label'),
+                                            '', li_info.get('url'))
+                urls.append(li_info.get('url'))
+
+        if self._mode is 'xbmc':
+            if override_view_mode:
+                skin = xbmc.getSkinDir()
+                thumbnail_view = THUMBNAIL_VIEW_IDS.get(skin)
+                if thumbnail_view:
+                    cmd = 'Container.SetViewMode(%s)' % thumbnail_view
+                    xbmc.executebuiltin(cmd)
+            xbmcplugin.addDirectoryItems(self.handle, items, len(items))
+            for id in sort_method_ids:
+                xbmcplugin.addSortMethod(self.handle, id)
+            xbmcplugin.endOfDirectory(self.handle, updateListing=is_update)
+
+        return urls
+
+    def _make_listitem(self, label, label2='', iconImage='', thumbnail='',
+                       path='', **options):
+        
+        li = xbmcgui.ListItem(label, label2=label2, iconImage=iconImage, thumbnailImage=thumbnail, path=path)
```

Comment:
`iconImage` and `thumbnailImage` are both deprecated. Please use `setArt` instead on the created Listitem